Utils/azure-explorer-common/src/com/microsoft/tooling/msservices/helpers/azure/sdk/AzureSDKManager.java

```diff
@@ -240,10 +254,77 @@ public static Resource createApplicationInsightsResource(@NotNull SubscriptionDe
                 .getAvailableGeoLocations();
     }
 
+    public static List<ApplicationInsightsComponent> getInsightsResources(@NotNull SubscriptionDetail subscription) throws IOException, RestOperationException {
+        final InsightsManager insightsManager = getInsightsManagerClient(subscription.getSubscriptionId());
+        return insightsManager == null ? null : insightsManager.components().list();
+    }
+
+    public static ApplicationInsightsComponent createInsightsResource(@NotNull SubscriptionDetail subscription,
+                                                             @NotNull String resourceGroupName,
+                                                             boolean isNewGroup,
+                                                             @NotNull String resourceName,
+                                                             @NotNull String location) throws Exception {
+        final InsightsManager insightsManager = getInsightsManagerClient(subscription.getSubscriptionId());
+        if (insightsManager == null) { // not signed in
+            return null;
+        }
+        final Azure azure = AuthMethodManager.getInstance().getAzureClient(subscription.getSubscriptionId());
+        if (isNewGroup) {
+            azure.resourceGroups().define(resourceGroupName).withRegion(location).create();
+        }
+        return insightsManager.components()
+                .define(resourceName)
+                .withRegion(location)
+                .withExistingResourceGroup(resourceGroupName)
+                .withApplicationType(ApplicationType.WEB)
+                .withKind("web")
+                .create();
+    }
+
+    // As AI SDK didn't provide list supported regions interface, call the api directly
```

Comment:
//TODO: AI SDK doesn't provide a method to list regions which are available regions to create AI, we are requiring the SDK to provide that API, before SDK side fix, we will use our own impl